base/src/main/java/butter/droid/base/data/DataModule.java

```diff
@@ -34,40 +33,39 @@
 import butter.droid.base.utils.StorageUtils;
 import dagger.Module;
 import dagger.Provides;
+import okhttp3.Cache;
+import okhttp3.OkHttpClient;
+import okhttp3.OkHttpClient.Builder;
 
 @Module
 public class DataModule {
 
-    @Provides @Singleton public OkHttpClient provideOkHttpClient(Context context) {
-        OkHttpClient client = new OkHttpClient();
-        client.setConnectTimeout(30, TimeUnit.SECONDS);
-        client.setReadTimeout(60, TimeUnit.SECONDS);
-        client.setRetryOnConnectionFailure(true);
-
+    @Provides @Singleton public Cache provideCache(Context context) {
         int cacheSize = 10 * 1024 * 1024;
         File cacheLocation = new File(
                 PrefUtils.get(context, Prefs.STORAGE_LOCATION, StorageUtils.getIdealCacheDirectory(context)
                         .toString()));
         cacheLocation.mkdirs();
-        com.squareup.okhttp.Cache cache = null;
-        try {
-            cache = new com.squareup.okhttp.Cache(cacheLocation, cacheSize);
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
-        client.setCache(cache);
+        return new Cache(cacheLocation, cacheSize);
+    }
 
-        return client;
+    @Provides @Singleton public OkHttpClient provideOkHttpClient(Cache cache) {
+        return new Builder()
+                .connectTimeout(30, TimeUnit.SECONDS)
+                .readTimeout(60, TimeUnit.SECONDS)
+                .retryOnConnectionFailure(true)
+                .cache(cache)
+                .build();
     }
 
-    @Provides @Singleton public OkHttpDownloader provideOkHttpDownloader(Context context) {
-        return new OkHttpDownloader(context);
+    @Provides @Singleton public OkHttp3Downloader provideOkHttpDownloader(OkHttpClient client) {
+        return new OkHttp3Downloader(client);
     }
 
-    @Provides @Singleton public Picasso providePicasso(Context context, OkHttpDownloader okHttpDownloader) {
-        Picasso.Builder builder = new Picasso.Builder(context);
-        builder.downloader(okHttpDownloader);
-        return builder.build();
+    @Provides @Singleton public Picasso providePicasso(Context context, OkHttp3Downloader okHttpDownloader) {
+        return new Picasso.Builder(context)
+                .downloader(okHttpDownloader)
```

Comment:
I have tested image loading and it seams to be working. First time images are loaded from web (red triangle), after that they are loaded from memory (green triangle). When app is killed  and reopened images are loaded from disk (blue triangle). That is expected behaviour so this can be mergerd.